File: Code/C#/HVPSAPI/Messages/IndicateStateMessage.cs
using Core.Messages.Messages;
using HVPSAPI.DataMemberNames.Messages;
using HVPSCore.Enums;
using Microsoft.Win32;
using System;
using System.Net;
using System.Runtime.Serialization;
using System.Text.Json.Serialization;
using HVPSCore.Enums;
using MessageTypes.Attributes;
using HVPSAPI.DataMemberNames.Requests;
namespace HVPSAPI.Messages
{
    [DataContract]
    public class IndicateStateMessage : TypedMessageBase
    {
        [JsonPropertyName(IndicateStateMessageDataMemberNames.SystemState)]
        [JsonInclude]
        [DataMember(Name = IndicateStateMessageDataMemberNames.SystemState)]
        public SystemState SystemState { get; protected set; }
        public IndicateStateMessage()
            : base()
        {
            Type = MessageTypes.IndicateState;
        }
    }
}

File: Code/JavaScript/client/src/messages/IndicateStateMessage.js
export default class IndicateStateMessage
{
 static toJSON(o){ 
    const r = {};
   r["tpe"]="is";
    r["s"]=o.systemState;
    return r;
 }
 static fromJSON(o){
    const r = {};
    r.systemState=o["s"];
   return r;
 }
}